cmake/api_macros.cmake

```diff
@@ -86,6 +86,20 @@ macro(_OPENDDS_GENERATE_EXPORT_MACRO_COMMAND  target  output)
   set(${output} ${_output_file})
 endmacro()
 
+macro(OPENDDS_MARK_AS_GENERATED target)
+
+  if (NOT TARGET ${target})
```

Comment:
Also check if the target is a shared library. If it's not, then there won't be an export header.

Reply:
I'm pretty sure this would run into the same problem where we cannot check if it is a shared library because that is a property set by the library which is in a seperate directory from the one including the library.

I tried a few different ways of doing this and was running into the same NOTFOUND issue as with GENERATED.

I realize the way it is currently doing it is inefficient, but I'm not sure there is a way around that.

Reply:
Okay, then this just seems like more motivation to investigate if we can just generate the export header at configure step instead of the build step. Out of curiosity I went ahead and tried this and it seems to solve the problem, at least for the test you came up with.

```diff
diff --git a/cmake/api_macros.cmake b/cmake/api_macros.cmake
index 2d3d688d5..2325be383 100644
--- a/cmake/api_macros.cmake
+++ b/cmake/api_macros.cmake
@@ -75,13 +75,15 @@ macro(_OPENDDS_GENERATE_EXPORT_MACRO_COMMAND  target  output)
     message(FATAL_ERROR "Failed to find required script 'generate_export_file.pl'")
   endif()
 
-  add_custom_command(
-    OUTPUT ${_output_file}
-    DEPENDS perl
+  execute_process(
     COMMAND ${CMAKE_COMMAND} -E env "DDS_ROOT=${DDS_ROOT}" "TAO_ROOT=${TAO_ROOT}"
-      $<TARGET_FILE:perl> ${_gen_script} ${target} $<ANGLE-R> ${_output_file}
-    VERBATIM
-  )
+      perl ${_gen_script} ${target}
+    OUTPUT_FILE ${_output_file}
+    RESULT_VARIABLE _export_script_exit_status)
+  if(NOT _export_script_exit_status EQUAL "0")
+    message(FATAL_ERROR
+      "Export header script for ${target} exited with ${_export_script_status}")
+  endif()
 
   set(${output} ${_output_file})
 endmacro()
```